src/PartitionLoops.cpp

```diff
@@ -0,0 +1,926 @@
+
+            // The steady state was simplified by
+            // FindSteadyState. It's not safe to simplify the prologue
+            // and epilogue at the same time - FindSteadyState uses
+            // already found simplifications to make new ones,
+            // including potentially using a simplification of the
+            // prologue end to assist the epilogue start. Instead we
+            // do a bounds-based simplification of the prologue and
+            // epilogue as a later pass.
+            //
+            // Note that the bounds we use are actually slightly
+            // conservative. E.g. we claim the prologue can run up
+            // till min_steady, where it actually runs to
+            // clamp(min_steady, loop_min, loop_max). Fortunately, the
+            // case where our bound is incorrect is the case where the
+            // prologue or epilogue have extent zero, so the incorrect
+            // simplification doesn't matter.
+            BoundsBasedSimplify simplify_prologue(op->name, op->min, min_steady.defined() ? min_steady - 1 : Expr());
+            BoundsBasedSimplify simplify_epilogue(op->name, max_steady, op->min + op->extent - 1);
+
+            if (make_prologue) {
+                // Clamp the prologue end to within the existing loop bounds,
+                // then pull that out as a let statement.
+                min_steady = clamp(min_steady, op->min, op->min + op->extent);
+                string min_steady_name = op->name + ".prologue";
+                lets.push_back(make_pair(min_steady_name, min_steady));
+                min_steady = Variable::make(Int(32), min_steady_name);
+            } else {
+                min_steady = op->min;
+            }
+
+            if (make_epilogue) {
+                // Clamp the epilogue start to be between the prologue
+                // end and the loop end, then pull it out as a let
+                // statement.
+                max_steady = clamp(max_steady, min_steady, op->min + op->extent);
+                string max_steady_name = op->name + ".epilogue";
+                lets.push_back(make_pair(max_steady_name, max_steady));
+                max_steady = Variable::make(Int(32), max_steady_name);
+            } else {
+                max_steady = op->extent + op->min;
+            }
+
+
+            debug(3) << "\nSimpler body: " << simpler_body << "\n";
+
+            Stmt new_loop;
+
+            if (op->for_type == ForType::Serial) {
+                // Steady state.
+                new_loop = For::make(op->name, min_steady, max_steady - min_steady,
+                                     op->for_type, op->device_api, simpler_body);
+
+                if (make_prologue) {
+                    Stmt prologue = For::make(op->name, op->min, min_steady - op->min,
+                                              op->for_type, op->device_api, body);
+                    //prologue = simplify_prologue.mutate(prologue);
```

Comment:
Why is this commented out? It looks like you are not applying the bounds simplification to either the prologue or the epilogue right now.